store/postgres/src/relational/index.rs

```diff
@@ -0,0 +1,848 @@
+//! Parse Postgres index definition into a form that is meaningful for us.
+use std::fmt::{Display, Write};
+
+use graph::itertools::Itertools;
+use graph::prelude::{
+    lazy_static,
+    regex::{Captures, Regex},
+    BlockNumber,
+};
+
+use crate::block_range::{BLOCK_COLUMN, BLOCK_RANGE_COLUMN};
+use crate::relational::{BYTE_ARRAY_PREFIX_SIZE, STRING_PREFIX_SIZE};
+
+use super::VID_COLUMN;
+
+#[derive(Debug, PartialEq)]
+pub enum Method {
+    Brin,
+    BTree,
+    Gin,
+    Gist,
+    Unknown(String),
+}
+
+impl Display for Method {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        use Method::*;
+        match self {
+            Brin => write!(f, "brin")?,
+            BTree => write!(f, "btree")?,
+            Gin => write!(f, "gin")?,
+            Gist => write!(f, "gist")?,
+            Unknown(s) => write!(f, "{s}")?,
+        }
+        Ok(())
+    }
+}
+
+impl Method {
+    fn parse(method: String) -> Self {
+        match Self::try_from(method.as_str()) {
+            Ok(m) => m,
+            Err(_) => Method::Unknown(method),
+        }
+    }
+}
+
+impl<'a> TryFrom<&'a str> for Method {
+    type Error = &'a str;
+
+    fn try_from(method: &'a str) -> Result<Self, Self::Error> {
+        use Method::*;
+
+        match method {
+            "brin" => Ok(Brin),
+            "btree" => Ok(BTree),
+            "gin" => Ok(Gin),
+            "gist" => Ok(Gist),
+            _ => Err(method),
+        }
+    }
+}
```

Comment:
Considering our input is borrowed, there is no need to return it as an `Error`. 
I couldn't find where we would use the `Error` value on the rest of this module, so I believe it would be safe just to use `()` here.

Also, there's a debate about whether to use `FromStr/str::parse()` instead of `TryFrom<&str>::try_from`, since the former would be more readable, but that's subjective.

Personally, I find `FromStr` more practical because it doesn't require an import line on every module that uses it.

Reply:
I agree - I changed this to use `FromStr`